### ISSUE TITLE
Email directly from server rather than using mailto: links

### ISSUE THREAD

**Issue by howardgrigg** [tbar0970/jethro-pmm#173]
This is a feature request/for discussion. 

It would be easier to use the email features if you were able to send email directly from the webpage using a form. 

The issue is that the mailto: links presume that the user has a mail client configured and working on their computer/tablet/phone. Many situations arise where this isn't the case such as those who use webmail exclusively (myself) and also when using a shared computer (such as the service computer).

An alternative which I think would be preferable would be for those links to instead open a modal window with a form to complete the email contents. Then send the email directly from the server. This would allow easier communication directly from inside Jethro. It wouldn't require an overly complex email setup and would most likely be able to be handled with php's native email function.

**Comment by tbar0970** [comment]
Good suggestion.  It's come up before.  It would be particularly valuable when you want to send to more than ~20 recipients where the current batching approach is somewhat clunky.

The challenge here is that we'd be building a mail client into Jethro, and my experience shows that mail clients are not as simple as you'd think!  The really basic text-only message is straightforward enough, but once you start getting into attachments, inline images etc it starts getting harder.  Also, SPF/DKIM and spam detection need to be considered - if an email that's "from" tom@mychurch.com has been dispatched from a server at somejethrosystem.com some recipient servers will think it's spam.  There are ways around that but it requires some careful configuration.

BTW you can configure your browser to open mailto links in webmail - google will reveal how to do so.  The tricky thing is whether or not Jethro opens a new tab/window for webmail or not - there could be some further tweaking we could do there.

**Comment by howardgrigg** [comment]
Yea there's certainly issues that would need to be considered but I don't think they would be insurmountable. We could look at using preexisting frameworks - in fact I see we've already got the swiftmailer library as part of the app. That could easily handle the simple uses we would have. 

It may increase installation complexity although most servers have mail setup. Avoiding spam folders is always a concern - generally I haven't had issues as long as the domains match - you can utilise the "reply-to" field instead of the "from" field to avoid having all the replies sent to the server.

**Comment by andrewfn** [comment]
mailgun has a beautifully simple api for sending email, and their deliverabllity is excellent.
The main issue would be whom the email is sent from because of spf record checking. One solution is to give the user the option of sending from their own account, or from a built-in Jethro account that can be configured on a domain that allows mailgun to send email. This is what CiviCRM does and it works well.

**Comment by tbar0970** [comment]
Yes, there are a few services like mailgun that would be valuable.  Jethro would need to provide a user interface which calls an email service.

Having the options to use either your own mail account (via mailto) or built-in mailer would be good.

**Comment by tim-pearce** [comment]
Email from Jethro is something I would like too.
I haven't looked into the complexity involved in incorporating it into something like Jethro but there is Contact Form 7 in Wordpress which I have been using.
I'm thinking in terms of having forms/templates which you set up for different purposes (including a 'general' one).

**Comment by tbar0970** [comment]
Along with this, it would be valuable to be able to merge in recipients' details.  Along with the obvious "Dear %firstname%" scenario, it could be used to say "We currently have mobile number %mobile_tel% on file; before we publish the directory please confirm if this is correct".
